### ISSUE TITLE
on_after_backward docs for logging histograms

### ISSUE THREAD

**Issue by Data-drone** [PyTorchLightning/pytorch-lightning#5525]
#### What is your question?

I was following the example `on_after_backward` code for logging parameter variables for tensorboard histograms.
https://pytorch-lightning.readthedocs.io/en/latest/_modules/pytorch_lightning/core/hooks.html#ModelHooks.on_after_backward

#### Code

    # example to inspect gradient information in tensorboard
    if self.trainer.global_step % 25 == 0:  # don't make the tf file huge
        params = self.state_dict()
        for k, v in params.items():
            self.logger.experiment.add_histogram(
                tag=k, values=v.grad, global_step=self.trainer.global_step
                )

and I was getting errors with `v.grad` in my `state_dict` `v.grad` is returning NoneTypes rather than valid tensors for all tensors?

Has the API for pytorch changed perhaps? or is this method mainly just suitable for logging the values of the variables instead of the gradients?

#### What's your environment?

 - OS: Ubuntu 20.04
 - Packaging pip
 - Version 1.1.4
pytorch 1.7.0

**Comment by rohitgr7** [comment]
yeah just the example is a bit incorrect since `state_dict` doesn't contain the parameters, so no grad. should be replaced with:
```python
params = self.named_parameters()
    for k, v in params:
    ....
```
mind send a PR/

**Comment by tchaton** [comment]
Hey @Data-drone,

I think @rohitgr7 found the error in your code example.

I will be closing this issue. Feel free to open it if you have more questions.

Best,
T.C

**Comment by rohitgr7** [comment]
@tchaton the error is in docs.. need to fix it.
